### PR TITLE
(vsphere-iso) add the ability to check if the IP is within the an IP range

### LINKED ISSUE #9349: How to disable packer to get ipv6 address while connecting via SSH?
Hi team,

I am using packer vSphere builder to build images from a remote vCenter. VMs will have two IPs: IPv4 and IPv6. I found that it seems like packer will randomly pick one of them - and this may lead problem. 
```
...

==> vsphere: Power on VM...
==> vsphere: Waiting 10s for boot...
==> vsphere: HTTP server is working at http://10.xx.xx.xx:8495/
==> vsphere: Typing boot command...
==> vsphere: Waiting for IP...
==> vsphere: IP address: 2620:124:6020:xxx:xxx:xxx:xxx:xxx
==> vsphere: Using ssh communicator to connect: 2620:124:6020:xxx:xxx:xxx:xxx:xxx
==> vsphere: Waiting for SSH to become available...
```
And if packer picks an IPv6 address, then it hangs there..

Is there an option to disable the IPv6 address picks up? or somewhere could be configurable?

### PR DESCRIPTION
**Pull request by jhawk28**
added a filed called "ip_wait_address" to allow packer to wait for an ip address that fits within the range (cidr address).

Closes #9349

### CODE REVIEW

**Comment by sylviamoss** [comment]
I think this looks good. What do you think about updating the `vmware/govmomi` version instead, as suggested by @jiatongw in https://github.com/hashicorp/packer/issues/9349? I think I like more his suggestions, both the template config and the implementation would be more simple.

**Comment by jhawk28** [comment]
I think it depends on what is trying to be fixed. Most people are seeing this as a expecting IPv4 address and the VM install is choosing an IPv6 address. Adding an isIPv4 flag would fix that. Here are some circumstances that the flag doesn't fix:

* expecting a IPv6 address and getting a IPv4
* getting an 169.x.x.x address on Windows
* getting an address from a NIC that is unresolvable (could happen with the multiple NICs now)

If we don't care about any of those, then a simple flag would suffice.

**Comment by sylviamoss** [comment]
Got it! I don't know much about all of the possibilities here but I agree with your solution now. Thanks for explaining! 
I'll review this again and try to run some tests.

**Comment by SwampDragons** [comment]
I like this -- my only request is that you update the docs comment to include what the default value is.

**Comment by jhawk28** [comment]
@SwampDragons what do we want the default value to be? We could make it default to nothing which matches the current behavior or we could make the default "0.0.0.0/0" to default to ipv4 only addresses.

**Comment by jiatongw** [comment]
@jhawk28 I would like to make it default to an IPv4 address. For IPv6, maybe could add a flag or something? 
the current behavior - default to nothing may cause unpredictable building failures if it picks an IPv6 address.

**Comment by SwampDragons** [comment]
yeah if most issues we're seeing are people who expect an ipv4 address, I think it probably makes sense to default it to that. We just need to make sure we tag this as a potential backwards incompatability in the changelog.

**Comment by jhawk28** [comment]
I think that people are currently seeing ipv4 issues right now because we have mostly ipv4 networks. In the semi-near future, we "should" see more ipv6 networks. But they have been trying to switch to ipv6 for a long time. I could update the variable to a *string. This would allow the default (nil) to be "0.0.0.0/0" for ipv4 filtering. If you set it to empty string (""), it would clear the filter for any address. ipv6 filtering of course can be done with "0:0:0:0:0:0:0:0/0".

**Comment by VladRassokhin** [comment]
Better later than never, what do you thinks about possible property values:

* `""` (empty string) — IPv4
* `"IPv4"` — IPv4
* `"IPv6"` — IPv6
* nil — IPv4 or IPv6
* `"any"` — IPv4 or IPv6
* IPv4 CIDR — IPv4 CIDR
* IPv6 CIDR — IPv4 CIDR

I think that 'IPv4' is much simpler to understand than `0.0.0.0/0` and for ipv6 that's even simpler.

**Comment by VladRassokhin** [comment]
And second questing why it called `ip_wait_address` when actually it's `ip_wait_cidr`?

File: builder/vsphere/common/step_wait_for_ip.go [hunk 1]
```diff
@@ -27,6 +28,14 @@ type WaitIpConfig struct {
 	// [ParseDuration](https://golang.org/pkg/time/#ParseDuration) documentation
 	//  for full details.
 	SettleTimeout time.Duration `mapstructure:"ip_settle_timeout"`
+	// Set this to a CIDR address to cause the service to wait for an address that is contained in
+	// this network range. Defaults to "0.0.0.0/0" for any ipv4 address. Examples include:
+	//
+	// * empty string ("") - remove all filters
+	// * "0:0:0:0:0:0:0:0/0" - allow only ipv6 addresses
+	// * "192.168.1.0/24 - only allow ipv4 addresses from 192.168.1.1 to 192.168.1.254
```
Comment (VladRassokhin) [inline builder/vsphere/common/step_wait_for_ip.go:36]: Missing quote after `/24`

**Pull request merged**